### ISSUE TITLE
Indentation check for statement-only lambda

### ISSUE THREAD

**Issue by pietern** [checkstyle/checkstyle#2184]
Follow up on #1548.

Indentation is not yet checked for statement-only lambda's like this:

```java
    foos.forEach(f -> //indent:4 exp:4
        IntStream //indent:8 exp:8
            .range(1, 4) //indent:12 exp:12
            .forEach(i -> f.bars.add("bar"))); //indent:12 exp:12
```

The `.range` and `.forEach` calls may have any indentation without any warning being emitted.

**Comment by rnveach** [comment]
@romani With 6.17 I am getting warnings on all lines in example if indentation is set to 0 (incorrect), so we are validating them now.

````
$ cat TestClass.java
public class TestClass {
    void method() {
foos.forEach(f ->
IntStream
.range(1, 4)
.forEach(i -> f.bars.add("bar")));
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
      <module name="Indentation">
          <property name="basicOffset" value="4"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-6.17-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:3: 'method def' child have incorrect indentation level 0, expected level should be 8. [Indentation]
[ERROR] TestClass.java:4: 'IntStream' have incorrect indentation level 0, expected level should be 4. [Indentation]
[ERROR] TestClass.java:5: '.' have incorrect indentation level 0, expected level should be 4. [Indentation]
[ERROR] TestClass.java:6: '.' have incorrect indentation level 0, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 4 errors.
````

**Comment by romani** [comment]
@rnveach , thanks a lot.
@pietern I am closing issue for now.

**Comment by pietern** [comment]
@romani Cool!
